ibm/service/power/resource_ibm_pi_network_interface.go

```diff
@@ -0,0 +1,356 @@
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		_, err = isWaitForIBMPINetworkPortUpdateAvailable(ctx, networkC, networkID, networkInterfaceID, instanceID, d.Timeout(schema.TimeoutUpdate))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+	}
+	d.SetId(fmt.Sprintf("%s/%s/%s", cloudInstanceID, networkID, networkInterfaceID))
+
+	return resourceIBMPINetworkInterfaceRead(ctx, d, meta)
+}
+
+func resourceIBMPINetworkInterfaceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	parts, err := flex.IdParts(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	networkC := instance.NewIBMPINetworkClient(ctx, sess, parts[0])
+	networkInterface, err := networkC.GetNetworkInterface(parts[1], parts[2])
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.Set(Attr_CRN, networkInterface.Crn)
+	d.Set(Attr_IPAddress, networkInterface.IPAddress)
+	d.Set(Attr_MacAddress, networkInterface.MacAddress)
+	d.Set(Attr_Name, networkInterface.Name)
+	d.Set(Attr_NetworkInterfaceID, networkInterface.ID)
+	d.Set(Attr_NetworkSecurityGroupID, networkInterface.NetworkSecurityGroupID)
+	if networkInterface.Instance != nil {
+		pvmInstance := []map[string]interface{}{}
+		instanceMap := pvmInstanceToMap(networkInterface.Instance)
+		pvmInstance = append(pvmInstance, instanceMap)
+		d.Set(Attr_Instance, pvmInstance)
+	} else {
+		d.Set(Attr_Instance, nil)
+	}
+	d.Set(Attr_Status, networkInterface.Status)
+	if networkInterface.Crn != nil {
+		tags, err := flex.GetTagsUsingCRN(meta, string(*networkInterface.Crn))
+		if err != nil {
+			log.Printf("Error on get of network interface (%s) pi_user_tags: %s", *networkInterface.ID, err)
+		}
+		d.Set(Arg_UserTags, tags)
+	}
+
+	return nil
+}
+
+func resourceIBMPINetworkInterfaceUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	parts, err := flex.IdParts(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	networkC := instance.NewIBMPINetworkClient(ctx, sess, parts[0])
+	body := &models.NetworkInterfaceUpdate{}
+
+	hasChange := false
+	if d.HasChange(Arg_UserTags) {
+		if crn, ok := d.GetOk(Attr_CRN); ok {
+			oldList, newList := d.GetChange(Arg_UserTags)
+			err := flex.UpdateGlobalTagsUsingCRN(oldList, newList, meta, crn.(string), "", UserTagType)
+			if err != nil {
+				log.Printf("Error on update of network interface (%s) pi_user_tags: %s", parts[2], err)
+			}
+		}
+	}
+	if d.HasChange(Arg_Name) {
+		name := d.Get(Arg_Name).(string)
+		body.Name = &name
+		if v, ok := d.GetOk(Arg_InstanceID); ok {
+			instanceID := v.(string)
+			body.InstanceID = &instanceID
+		}
+		hasChange = true
+	}
+	if d.HasChange(Arg_InstanceID) {
+		instanceID := d.Get(Arg_InstanceID).(string)
+		body.InstanceID = &instanceID
+		hasChange = true
+	}
```

Comment:
IMO it is ok to pass both instanceID and name in one shot, instead of having another variable called `hasChange`

Reply:
Yes, you are correct -- we only need to set name and instanceID if they have changed. Updated it.